tests/contracts/nonfungible-token/contract.scilla

```diff
@@ -0,0 +1,411 @@
+                    ownedTokenCount := new_otc
+                end;
+
+                
+
+                msg  = {_tag : Main; _recipient : _sender; _amount : Uint128 0; 
+                    code : code_success};
+                msgs = one_msg msg;
+                send msgs
+            | False =>
+                (* Unauthorized transaction *)
+                msg  = {_tag : Main; _recipient : _sender; _amount : Uint128 0; 
+                    code : code_not_authorized};
+                msgs = one_msg msg;
+                send msgs
+            end
+        end
+    end
+end
+
+(* @dev: Approves another address to transfer the given token ID                *)
+(* - There can only be one approved address per token at a given time           *)
+(* - Absence of entry in tokenApproval indicates there is no approved address   *)
+(* param: to Address to be approved for the given token id                      *)
+(* param: tokenId uint256 id of the token to be apporved                        *)
+
+
+transition approve(to: Address, tokenId: Uint256)
+
+    copy_tokenOwnerMap <- tokenOwnerMap;
+    copy_operatorApproval <- operatorApprovals;
+
+    (* Get tokenOwner Address *)
+    getTokenOwner = builtin get copy_tokenOwnerMap tokenId;
+    match getTokenOwner with
+    | None =>
+        (* Token not found *)
+        msg  = {_tag : Main; _recipient : _sender; _amount : Uint128 0; 
+                code : code_not_found};
+                msgs = one_msg msg;
+                send msgs
+    | Some tokenOwner =>
+        checkApprovedForAll = isApprovedForAll _sender tokenOwner copy_operatorApproval;
+        checkOwner = isTokenOwner _sender tokenId copy_tokenOwnerMap;
+        isAuthorized = orb checkApprovedForAll checkOwner;
+        match isAuthorized with
+        | True =>
+            (* add to token approval mapping *)
+            copy_tokenApprovals <- tokenApprovals;
+            new_tokenApprovals = builtin put copy_tokenApprovals tokenId to;
+            tokenApprovals := new_tokenApprovals;
+
+            msg  = {_tag : Main; _recipient : _sender; _amount : Uint128 0; 
+                code : code_success};
+            msgs = one_msg msg;
+            send msgs
+        | False =>
+            (* Unauthorized transaction *)
+            msg  = {_tag : Main; _recipient : _sender; _amount : Uint128 0; 
+                code : code_not_authorized};
+            msgs = one_msg msg;
+            send msgs
+        end
+    end
+end
+
+(* @dev: sets or unsets the approval of a given operator                *)
+(* @param: address to be set or unset as operator                       *)
+(* @param: approved - status of the approval to be set                  *)
+
+transition setApprovalForAll(to: Address, approved: Bool)
+
+    copy_operatorApproval <- operatorApprovals;
+    isValidOperation = builtin eq _sender to;
+    (* require _sender is not approving himself *)
+    match isValidOperation with
```

Comment:
The name `isValidOperation` is a bit misleading because you raise a failure when it is `True`.

Reply:
I fixed this by using a `negb`. Sorry for the confusion.